api_test/src/test/java/ecse429/group7/TestTodos.java

```diff
@@ -0,0 +1,320 @@
+
+    @Test
+    public void testGetTodosResponseSecondDoneStatus()
+    {
+        HttpResponse<JsonNode> response = Unirest.get("/todos").asJson();
+        assertEquals(response.getBody().getObject().getJSONArray("todos").getJSONObject(1).getString("doneStatus"), "false");
+    }
+
+    @Test
+    public void testGetTodosResponseDescriptionEmpty()
+    {
+        HttpResponse<JsonNode> response = Unirest.get("/todos").asJson();
+        assertEquals(response.getBody().getObject().getJSONArray("todos").getJSONObject(1).getString("description"), "");
+    }
+
+    // GET /todos/:id
+    @Test
+    public void testGetTodosInvalidTodoStatusCode()
+    {
+        assertGetStatusCode("/todos/3", STATUS_CODE_NOT_FOUND);
+    }
+
+    @Test
+    public void testGetTodosInvalidTodoErrorMessage()
+    {
+        assertGetErrorMessage("/todos/3", "Could not find an instance with todos/3", 0);
+    }
+
+    @Test
+    public void testGetTodosValidStatusCode()
+    {
+        assertGetStatusCode("/todos/2", STATUS_CODE_OK);
+    }
+
+    @Test
+    public void testGetTodosValidBodyTitle()
+    {
+        HttpResponse<JsonNode> response = Unirest.get("/todos/2").asJson();
+        assertEquals(response.getBody().getObject().getJSONArray("todos").getJSONObject(0).getString("title"), "file paperwork");
+    }
+
+    @Test
+    public void testGetTodosValidTaskOf()
+    {
+        HttpResponse<JsonNode> response = Unirest.get("/todos/2").asJson();
+        assertEquals(response.getBody().getObject().getJSONArray("todos").getJSONObject(0)
+            .getJSONArray("task-of").getJSONObject(0).getInt("id"), 1);
+    }
+
+    // HEAD /todos
+    @Test
+    public void testHeadTodos()
+    {
+        assertHeadStatusCode("/todos", STATUS_CODE_METHOD_NOT_ALLOWED);
+    }
+
+    // POST /todos
+    public  void testPostTodosInvalidStatusCode()
+    {
+        HttpResponse<JsonNode> response = Unirest.post("/todos").asJson();
+        assertEquals(response.getStatus(), STATUS_CODE_BAD_REQUEST);
+    }
+
+    @Test
+    public void testPostTodosErrorMessage()
+    {
+        HttpResponse<JsonNode> response = Unirest.post("/todos").asJson();
+        assertEquals(response.getBody().getObject().getJSONArray("errorMessages").getString(0), "title : field is mandatory");
+    }
+
+    @Test
+    public void testPostTodosJSONValidStatusCode()
+    {
+        HttpResponse<JsonNode> response = Unirest.post("/todos").body("{\"title\":\"NewTodo\"}").asJson();
+        
+        //Delete object created to reset state of database
+        int id = response.getBody().getObject().getInt("id");
+        Unirest.delete("/todos/" + String.valueOf(id)).asJson();
+        
+        assertEquals(response.getStatus(), STATUS_CODE_CREATED);
+    }
+
+    @Test
+    public void testPostTodosJSONValidTitle()
+    {
+        HttpResponse<JsonNode> response = Unirest.post("/todos").body("{\"title\":\"NewTodo\"}").asJson();
+        
+        //Delete object created to reset state of database
```

Comment:
Should this be a teardown function that can be added after tests?

Reply:
I was thinking about this but I don't know how to make it work. The problem is that the teardown depends on the ID that we get when we do the POST so I haven't figured this out